superset-frontend/src/SqlLab/components/ResultSet.tsx

```diff
@@ -501,36 +518,104 @@ export default class ResultSet extends React.PureComponent<
     return <div />;
   }
 
+  onAlertClose = () => {
+    this.setState({ alertIsOpen: false });
+  };
+
   renderRowsReturned() {
-    const { results, rows, queryLimit } = this.props.query;
+    const { results, rows, queryLimit, limitingFactor } = this.props.query;
+    let limitMessage;
     const limitReached = results?.displayLimitReached;
+    const isAdmin = !!this.props.user?.roles.Admin;
+    const displayMaxRowsReachedMessage = {
+      withAdmin: t(
+        `The number of results displayed is limited to %(rows)d by the configuration DISPLAY_MAX_ROWS.
+        Please add additional limits/filters or download to csv to see more rows up to the
+        the %(queryLimit)d limit.`,
+        { rows, queryLimit },
+      ),
+      withoutAdmin: t(
+        `The number of results displayed is limited to %(rows)d.
+        Please add additional limits/filters, download to csv, or contact an admin
+        to see more rows up to the the %(queryLimit)d limit.`,
+        { rows, queryLimit },
```

Comment:
one other thing I found is that multiline strings don't work well with the translations. I'd recommend breaking this into 3-4 strings.

Reply:
oh interesting, ok I will